pkg/clusters/addons/kong/builder.go

```diff
@@ -36,6 +39,18 @@ func (b *Builder) WithPostgreSQL() *Builder {
 	return b
 }
 
+func (b *Builder) WithEnterprise(dbMode DBMode) *Builder {
+	b.dbmode = dbMode
```

Comment:
As we've been discussing: `dbmode` doesn't really make sense here, the caller can use `WithPostgres` if they want to enable psql, or `WithDBLESS` if they want dbless.

```suggestion
func (b *Builder) WithEnterprise() *Builder {
```

Reply:
eeff8d91ce826dd138626da30da6711a030c8649